recipes/nextcloud.rb

```diff
@@ -0,0 +1,58 @@
+#
+# Cookbook:: osl-nextcloud
+# Recipe:: default
+#
+# Copyright:: 2022-2023, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+dbcreds = data_bag_item('proj-seagl', 'dbcredentials')
+servercreds = data_bag_item('proj-seagl', 'servercredentials')
+nextcloudcreds = data_bag_item('proj-seagl', 'nextcloudcredentials')
+
+node.override['percona']['version'] = '5.7'
+node.default['percona']['server']['root_password'] = servercreds['root_password']
+node.default['percona']['backup']['password'] = servercreds['backup_password']
+
+include_recipe 'osl-mysql::server'
+
+percona_mysql_database dbcreds['db_dbname'] do
+  password node['percona']['server']['root_password']
+end
+
+percona_mysql_user dbcreds['db_user'] do
+  database_name dbcreds['db_dbname']
+  password dbcreds['db_passw']
+  host dbcreds['db_host']
+  ctrl_password node['percona']['server']['root_password']
+  action [:create, :grant]
+end
```

Comment:
We will be using our production mysql cluster for this and _not_ hosting it on the same system. Please remove this. I recommend you try out the new `osl_mysql_test` resource for setting up the test server for this kitchen test.